test/generator.test.js

```diff
@@ -0,0 +1,526 @@
+        yield item;
+    }
+}
+
+async function sleepReturnValue(sleepMs, value) {
+    await sleep(sleepMs);
+    return value;
+}
+
+describe('generator', function() {
+    describe('mapconcurrent', function() {
+        describe('checkAddBatch', function() {
+            it('empty', function() {
+                let callbackInvoked = false;
+                const batch = [];
+                const result = checkAddBatch({
+                    checkAddBatch: () => {
+                        callbackInvoked = true;
+                        return true;
+                    }
+                }, batch, 1, 1);
+
+                assert.strictEqual(result, true);
+                assert.strictEqual(callbackInvoked, false);
+            });
+            it('full', function() {
+                let callbackInvoked = false;
+                const batch = [ 1 ];
+                const result = checkAddBatch({
+                    checkAddBatch: () => {
+                        callbackInvoked = true;
+                        return true;
+                    }
+                }, batch, 2, 1);
+
+                assert.strictEqual(result, false);
+                assert.strictEqual(callbackInvoked, false);
+            });
+            it('no-checkaddbatch', function() {
+                const batch = [ 1 ];
+                const result = checkAddBatch({ }, batch, 2, 2);
+                assert.strictEqual(result, true);
+            });
+            it('checkaddbatch-false', function() {
+                let callbackInvoked = false;
+                const batch = [ 1 ];
+                const result = checkAddBatch({ 
+                    checkAddBatch: () => {
+                        callbackInvoked = true;
+                        return false;
+                    }
+                }, batch, 2, 2);
+                assert.strictEqual(result, false);
+                assert.strictEqual(callbackInvoked, true);
+            });
+            it('checkaddbatch-true', function() {
+                let callbackInvoked = false;
+                const batch = [ 1 ];
+                const result = checkAddBatch({ 
+                    checkAddBatch: () => {
+                        callbackInvoked = true;
+                        return true;
+                    }
+                }, batch, 2, 2);
+                assert.strictEqual(result, true);
+                assert.strictEqual(callbackInvoked, true);
+            });
+        });
+        describe('executeBatch', function() {
+            it('empty-array', async function() {
+                const result = await executeBatch({
+                    execute: async function*(values) {
+                        for await (const value of values) {
+                            yield value + 1;
+                        }
+                    }
+                }, [ ]);
+                assert.deepStrictEqual(result, [ ]);
+            });
+            it('array', async function() {
```

Comment:
In the code, I didn't see checks that arrays are used. What if it gets Sets or Maps?

Reply:
That should be fine, as long as `for await (const v of items) { }` works which is true for Set or Map.